wire-runtime/src/main/java/com/squareup/wire/Message.java

```diff
@@ -69,6 +71,39 @@ protected final Object writeReplace() throws ObjectStreamException {
     return new MessageSerializedForm<>(this, (Class<Message>) getClass());
   }
 
+  /** Returns a new builder for this class, initialized with the data in this message. */
+  public final <B extends Builder<T, B>> B newBuilder() {
```

Comment:
Downside is you'll double-initialize the builder fields, once with a `newMutableList()` allocation only to overwrite with the version from `copyOf`. Perhaps `copyOf` could peek inside the `MutableOnWriteList` and replace its immutable list with the passed in version to avoid the extra allocation.